crypto/x509/x509_vfy.c

```diff
@@ -741,7 +742,13 @@ static int check_id(X509_STORE_CTX *ctx)
         if (!check_id_error(ctx, X509_V_ERR_IP_ADDRESS_MISMATCH))
             return 0;
     }
-    return 1;
+
+    if (vpm->purpose != X509_PURPOSE_SSL_SERVER ||
+	(vpm->flags & X509_V_FLAG_ALLOW_NO_SUBJECT_CHECK))
```

Comment:
Indentation appears off here

Reply:
put the ``||`` on the second line and indent that line an extra four.